site/app/controllers/MiscController.php

```diff
@@ -58,4 +72,197 @@ private function display_file() {
             }
         }
     }
+
+    private function downloadFile() {
+        //Additional security
+        if (!($this->core->getUser()->accessGrading())) {
```

Comment:
As Matt mentioned in the comments, this is not yet sufficient for security (although it is a step in the right direction). Although students are (presumably) no longer capable of downloading files, an instructor can still download arbitrary system files, including system configuration and assignment configuration/submissions from other courses. The best way to fix this is likely to treat the provided path as subdirectory of some safe directory - in conjunction with the restriction on `.` and `..` in paths, this should prevent the most blatant abuses.